src/index.ts

```diff
@@ -130,6 +135,20 @@ export class Tado {
     };
 
     this.#accessToken = await client.getToken(tokenParams);
+
+    if (this.#firstLogin) {
+      try {
+        const me = await this.getMe();
+        if (me.homes.length > 0) {
+          const home_id = me.homes[0].id;
+          const home = await this.getHome(home_id);
+          this.#isX = home.generation == "LINE_X";
+        }
+      } catch (err) {
+        console.error(`Could not determine TadoX status: ${err}`);
+      }
+      this.#firstLogin = false;
+    }
```

Comment:
Is this a safe assumption? It is likely that one account might end up with 2 homes (one legacy and one x) for a while. Might be unsafe to assume generation at the instance level. Could be better to have an private cache mapping home id to generation.

Alternatively allow providing generation to constructor?

Reply:
Good call. I was aiming to simplify for the end user but maybe adding an explicit `isX` at the end of the constructor is a good idea. I also like the idea of keeping an internal map. I'll implement both, with the constructor overriding the map.

Reply:
I've implemented this now

Reply:
Depending on how long the user has the `Tado` object alive for I guess it is feasible a house is added or changes during the lifetime of the object so maybe adding an explicit `updateTadoXStatus` call or having it check the status every token refresh wouldn't be crazy.